### ISSUE TITLE
ICE when including cigar.hpp with validators.hpp

### ISSUE THREAD

**Issue by smehringer** [seqan/seqan3#1194]
### Platform

  - SeqAn version: current master
  - Operating system: linux
  - Compiler: GCC-7.4, GCC-8.3 (GCC9 works)

### Description

Including both cigar.hpp and validators.hpp causes an ICE.

Seems to be related to the debug stream.

#### How to repeat the problem

```cpp
#include <seqan3/alphabet/cigar/cigar.hpp>
#include <seqan3/argument_parser/validators.hpp>

int main()
{}
```

**Comment by marehr** [comment]
Okay I found the problem and I think I have a fix. This is issue was funny, because they were two separate issues xD

One that had to do with the short-cut syntax of `Sentinel<>` that hide visibility errors, such as members being private/protected, but being accessed outside of the allowed scope.

And one ICE that has to do with `std::basic_ostream` and ranges.

**Comment by marehr** [comment]
Just for reference I reported it upstream https://github.com/ericniebler/range-v3/issues/1105 and it is now fixed upstream. I'll still add a PR to workaround it.

**Comment by h-2** [comment]
Thanks for tracking this down!

I will update the range-v3 branch soon. So if that already fixes  everything for SeqAn, we don't need the other changes?

**Comment by marehr** [comment]
yeah.

:sob: I... put so much work in it!!! So I thought... I can still... do the PR :crying_cat_face: :cry: 

(I'm just to good and fixed the problem at the root :sunglasses: )

The PR would add regression tests + original workaround.

**Comment by h-2** [comment]
> yeah.
> sob I... put so much work in it!!! So I thought... I can still... do the PR crying_cat_face cry

I understand that!

> (I'm just to good and fixed the problem at the root sunglasses )

It's the best kind of solution :+1:

> The PR would add regression tests + original workaround.

I am reluctant about adding a workaround for something that doesn't need working around, though..

**Comment by marehr** [comment]
Without workaround?

**Comment by smehringer** [comment]
Fixed by #1267
